src/main/java/org/alfresco/repo/domain/propval/AbstractPropertyValueDAOImpl.java

```diff
@@ -784,6 +784,12 @@ public Serializable getValueKey(Serializable value)
                 // It will be Serialized, so no search key
                 return null;
             }
+            else if (value instanceof Map<?, ?> && ((Map) value).isEmpty())

```

Comment:
Type argument is not required, Map  instead of Map<?, ?> would be sufficient.

Reply:
You are correct. The wildcard type parameter here doesn't buy us anything, but I wen't with it to keep to the existing style already present in this and surrounding classes.